plugins/inputs/tail/tail.go

```diff
@@ -260,7 +267,22 @@ func (t *Tail) tailNewFiles() error {
 			continue
 		}
 
-		for _, file := range g.Match() {
+		// Work around an issue in the doublestar library that requires read
+		// permissions on the directory to glob files (see
+		// https://github.com/bmatcuk/doublestar/issues/103).
+		// However, if the given file does not require globbing we can try to
+		// keep the file directly.
+		matches := g.Match()
+		if len(matches) == 0 {
+			if _, err := os.Lstat(filepath); err == nil {
+				matches = append(matches, filepath)
+			} else if !t.nomatch[filepath] {
+				t.Log.Warnf("No file(s) matching pattern %q! Make sure Telegraf has read permissions on directories followed by wildcards!", filepath)
```

Comment:
This warning is shown even when: The file doesn't exist (not a permission issue) and There are no wildcards in the pattern

The message should be more general or check if the pattern contains wildcards before suggesting permission issues. How about:

```suggestion
				t.Log.Warnf("No file(s) matching pattern %q: %v", filepath, err)
```